internal/provider/cloudexport_schema.go

```diff
@@ -428,18 +435,18 @@ func resourceDataToCloudExport(d *schema.ResourceData) (*models.CloudExport, err
 			export.Properties = &ibm
 		}
 	default:
-		return nil, fmt.Errorf("cloud_provider should be one of [aws, azure, ibm, gce], got: %q", cloudProvider)
+		return fmt.Errorf("cloud_provider should be one of [aws, azure, ibm, gce], got: %q", cloudProvider)
 	}
-	if cloudProvider == "bgp" {
-		bgp := models.BGPProperties{
-			UseBGPDeviceID: providerMap["use_bgp_device_id"].(string),
-			DeviceBGPType:  providerMap["device_bgp_type"].(string),
-		}
-		applyBGP := providerMap["apply_bgp"]
-		if v, ok := applyBGP.(bool); ok {
-			bgp.ApplyBGP = &v
-		}
-		export.BGP = &bgp
+	return nil
+}
+
+func resourceDataToBGPProperties(export *models.CloudExport, m map[string]interface{}) {
```

Comment:
```suggestion
func resourceDataToBGPProperties(data interface{}) (<bgp>, error) {}
```
The same as above: please modify the function to return bgp. Example conversion function for nested object: https://github.com/kentik/terraform-provider-kentik-synthetics/blob/c9ba2f17883e7a3e04cc504a18a85d7d122f5f9c/synthetics/test_conversions.go#L511
Function parameters might look a little bit different, of course.

Reply:
Done